lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -80,11 +75,10 @@ abstract class AbstractHydrator
     /**
      * Initializes a new instance of a class derived from <tt>AbstractHydrator</tt>.
      */
-    public function __construct(EntityManagerInterface $em)
+    public function __construct(protected EntityManagerInterface $_em)
```

Comment:
Let's not pollute the public API with those unfortunate underscores.

Reply:
Wasn't the public API already polluted by this (there was a protected property with that name). Unless we are taking named properties into account, the public API is not more polluted than it already was. But maybe we should stop polluted the public API with those, as an additional BC-break?

Reply:
The underscore can be removed in a separate PR to keep this PR small.

Reply:
My point is that the underscore is being _added_ to the constructor parameter with this PR. We should avoid that. If that means, we cannot use CPP yet for this parameter, so be it.